### PR TITLE
test wait_all_tasks_blocked with a cancelled task

### PR DESCRIPTION
**Pull request by graingert**

### CODE REVIEW

File: tests/test_debugging.py [hunk 1]
```diff
@@ -87,3 +88,19 @@ async def agen_func():
...
+async def test_wait_all_tasks_blocked_cancelled_task():
+    done = False
+
+    async def self_cancel(*, task_status):
+        task_status.started()
+        with move_on_after(-1):
+            await Event().wait()
+        nonlocal done
+        done = True
+
```
Comment (agronholm) [inline tests/test_debugging.py:102]: More in line with the rest of the code base.
```suggestion
    async def self_cancel(*, task_status):
        nonlocal done
        task_status.started()
        with move_on_after(-1):
            await Event().wait()

        done = True

    done = False
```

File: src/anyio/_backends/_asyncio.py [hunk 2]
```diff
@@ -1770,13 +1770,14 @@ def get_running_tasks() -> List[TaskInfo]:
 
 
 async def wait_all_tasks_blocked() -> None:
+    await checkpoint()
```
Comment (agronholm) [inline src/anyio/_backends/_asyncio.py:1773]: Why is this necessary?

**Review by agronholm — CHANGES_REQUESTED** [review]
One question and a bunch of code style changes.

File: src/anyio/_backends/_asyncio.py (hunk 2 above)
Comment (graingert) [reply src/anyio/_backends/_asyncio.py:1773]:
```
async def test_wait_all_tasks_blocked_cancelled_task_trio():
    with trio.testing.assert_checkpoints():
        await wait_all_tasks_blocked()
```

the trio version checkpoints

**Review by agronholm — APPROVED** [review]
